### PR TITLE
feat(planner): Enable new planner by default

### LINKED ISSUE #3747: Migrate to new planner framework
The new planner framework was introduced in #2983 .

Here's the tracking list of migration tasks from original planner to new planner:
- #3748 
- #3749 
- Scalar expression binding: #1537 #1538
- #5134
- Add a setting item to switch between planners #4989 
- #5163
- #5210
- #5203
- #5202
- #5235
- https://github.com/datafuselabs/databend/issues/5240
- #5259 
- #5343
- #5350
- #5353 
- #5360
- #5367
- #5351 
- https://github.com/datafuselabs/databend/issues/5536
- https://github.com/datafuselabs/databend/issues/5647
- https://github.com/datafuselabs/databend/issues/5651
- #5773

### PR DESCRIPTION
**Pull request by leiysky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Close #3747

### CODE REVIEW

**Review by xudong963 — COMMENTED** [review]
https://github.com/datafuselabs/databend/pull/6869/files#diff-c984ff53ad778ca83c9e974ba94c3323d0f33ce586d8087be0839c763454dc4e. The empty file seems useless

File: tests/logictest/suites/gen/02_function/02_0012_function_datetimes [hunk 1]
```diff
@@ -569,55 +569,55 @@ statement error 1079
...
 statement query T
-select addYears(to_date(18321), cast(-1, INT16));
+select addYears(to_date(18321), -1);
 
 ----
 2019-02-28
 
 statement query T
-select addYears(to_date(18321), cast(-1, INT32));
+select addYears(to_date(18321), -1);
```
Comment (sundy-li) [inline tests/logictest/suites/gen/02_function/02_0012_function_datetimes:608]:
```

databend > set enable_planner_v2 = 1;
Query OK, 0 rows affected (0.013 sec)

databend > select addYears(to_date(18321), cast(-1, INT64));
+----------------------------------------------+
| addYears(to_date(18321), CAST(- 1 AS Int64)) |
+----------------------------------------------+
| 2275-02-28                                   |
+----------------------------------------------+
1 row in set (0.015 sec)
```

Why this not work in plan_v2 ?

Comment (leiysky) [reply tests/logictest/suites/gen/02_function/02_0012_function_datetimes:608]: There was a bug just fixed in #6879

**Comment by sundy-li** [comment]
We should wait for #6256 merged

**Comment by Xuanwo** [comment]
#6478 has been merged.

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
It seems we should wait #7012 to merge.

**Comment by BohuTANG** [comment]
7012 merged, but makes this PR conflicting files

**Comment by leiysky** [comment]
It's ready now. @BohuTANG

**Review by sundy-li — APPROVED** [review]

**Review by andylokandy — APPROVED** [review]

**Pull request merged**